plugin-desktopswitch/desktopswitch.cpp

```diff
@@ -185,56 +179,23 @@ void DesktopSwitch::refresh()
     }
 }
 
-bool DesktopSwitch::isWindowHighlightable(WId window)
+bool DesktopSwitch::isWindowHighlightable(WId)
 {
-    // this method was borrowed from the taskbar plugin
-    QFlags<NET::WindowTypeMask> ignoreList;
-    ignoreList |= NET::DesktopMask;
-    ignoreList |= NET::DockMask;
-    ignoreList |= NET::SplashMask;
-    ignoreList |= NET::ToolbarMask;
-    ignoreList |= NET::MenuMask;
-    ignoreList |= NET::PopupMenuMask;
-    ignoreList |= NET::NotificationMask;
-
-    KWindowInfo info(window, NET::WMWindowType | NET::WMState, NET::WM2TransientFor);
-    if (!info.valid())
-        return false;
-
-    if (NET::typeMatchesMask(info.windowType(NET::AllTypesMask), ignoreList))
-        return false;
-
-    if (info.state() & NET::SkipTaskbar)
-        return false;
-
-    auto *x11Application = qGuiApp->nativeInterface<QNativeInterface::QX11Application>();
-    Q_ASSERT_X(x11Application, "DesktopSwitch", "Expected X11 connection");
-    WId appRootWindow = XDefaultRootWindow(x11Application->display());
-
-    // WM_TRANSIENT_FOR hint not set - normal window
-    WId transFor = info.transientFor();
-    if (transFor == 0 || transFor == window || transFor == appRootWindow)
-        return true;
-
-    info = KWindowInfo(transFor, NET::WMWindowType);
-
-    QFlags<NET::WindowTypeMask> normalFlag;
-    normalFlag |= NET::NormalMask;
-    normalFlag |= NET::DialogMask;
-    normalFlag |= NET::UtilityMask;
-
-    return !NET::typeMatchesMask(info.windowType(NET::AllTypesMask), normalFlag);
+    // Backend should emit signals only for higlightable windows and ignore others
+    // TODO: check
+    return true;
```

Comment:
This means a regression under X11; doesn't it?

Reply:
It should not change behavior. Backend just ignores non-highlightable windows so there is not need to filter them again.
See [LXQtTaskbarX11Backend::acceptWindow()](https://github.com/lxqt/lxqt-panel/blob/8b78adda8784fe52e3d5a4154cf71aef71f018ab/panel/backends/xcb/lxqttaskbarbackend_x11.cpp#L117)

Reply:
We could entirely remove the functiom

Reply:
You'll decide what to to do here. A part of code reviewing is to make sure that this PR won't cause any problem in the *current* functionality. All I'm saying is that the PR seems to cause a regression in this case, under X11.

Reply:
Oh, very sorry! It was GitHub's fault: it didn't load a file; so, I couldn't search in it. I saw that function after downloading the branch (which I should have done from the start).

Please remove the redundant function(s).